decorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class Decorator
+  def initialize(nameable_obj)
+    super()
+    @nameable_obj = nameable_obj
+  end
+
+  def correct_name
+    @nameable_obj.correct_name
+  end
+end
```

Comment:
currently, your decorator class does not inherit from your `Nameable` please check this and try again 😸